src/libaktualizr/uptane/uptane_network_test.cc

```diff
@@ -139,6 +140,44 @@ TEST(UptaneNetwork, DownloadFailure) {
   EXPECT_TRUE(result.first);
 }
 
+/*
+ * Output a log when connectivity is restored.
+ */
+TEST(UptaneNetwork, LogConnectivityRestored) {
+  //  testing::internal::CaptureStdout();
+  RecordProperty("zephyr_key", "OTA-991,TST-158");
+  TemporaryDirectory temp_dir;
+  conf.storage.type = StorageType::kSqlite;
+  conf.storage.path = temp_dir.Path();
+  conf.provision.expiry_days = "logconnect";
+  conf.provision.primary_ecu_serial = "logconnect";
+  const std::string good_url = conf.provision.server;
+  bool result;
+  auto http = std::make_shared<HttpClient>();
+  auto store = INvStorage::newStorage(conf.storage);
+  {
+    KeyManager keys(store, conf.keymanagerConfig());
+    Initializer initializer(conf.provision, store, http, keys, {});
+    result = initializer.isSuccessful();
+  }
+  EXPECT_TRUE(result);
+  {
+    conf.provision.server = conf.provision.server.substr(conf.provision.server.size() - 2) + "11";
+    KeyManager keys(store, conf.keymanagerConfig());
+    Initializer initializer(conf.provision, store, http, keys, {});
+    result = initializer.isSuccessful();
```

Comment:
I like this idea of feigning a broken connection by changing the server URL. However, does it require re-initialization every time? I was hoping we could just test this with calls to `CheckUpdates`. At any rate, we almost certainly can avoid extra instantiations of `KeyManager`.

Reply:
> At any rate, we almost certainly can avoid extra instantiations of KeyManager.

But `Initializer initializer(conf.provision, store, http, keys, {});` need `keys`. And from `doTestInit` it looks re-initialization is needed. I'm not sure though.

Reply:
I'm fairly confident that you can reuse the keys if not the entire initialization. To simulate a more realistic scenario, we should only initialize once and merely call `CheckUpdates` multiple times under different network conditions.

Reply:
> we should only initialize once and merely call CheckUpdates multiple times under different network conditions.

How can I get under different network conditions if I don't use initialize? I didn't find the function I need.